### PR TITLE
Merge "Google fixes"

### PR DESCRIPTION
**Pull request by adamnovak**
This will close #4196 by merging the changes from there.

I'm not sure if they exactly *work* though.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Support for a `--network` for `toil launch-cluster` for Google cloud (#4196)
 * Support for a `--use_private_ip` for `toil launch-cluster` to dial nodes by private IP instead of public IP (#4196)

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Review by DailyDreaming — APPROVED** [review]
Looks reasonable to me.

File: src/toil/provisioners/aws/awsProvisioner.py [hunk 1]
```diff
@@ -309,6 +310,9 @@ def launchCluster(self,
         :return: None
         """
 
+        if 'network' in kwargs:
+            logger.warning('AWS provisioner does not support a network parameter. Ignoring!')
```
Comment (mr-c) [inline src/toil/provisioners/aws/awsProvisioner.py:314]:
```suggestion
            logger.warning('AWS provisioner does not support a network parameter. Ignoring %s!', kwargs["network"])
```

File: src/toil/provisioners/gceProvisioner.py [hunk 2]
```diff
@@ -127,14 +127,18 @@ def launchCluster(self, leaderNodeType, leaderStorage, owner, **kwargs):
...
+        use_private_ip: even though a public ip exists, ignore it (optional)
         """
         if 'keyName' not in kwargs:
             raise RuntimeError("A keyPairName is required for the GCE provisioner.")
         self._keyName = kwargs['keyName']
         if 'botoPath' in kwargs:
             self._botoPath = kwargs['botoPath']
         self._vpcSubnet = kwargs['vpcSubnet'] if 'vpcSubnet' in kwargs else None
+        self._network = kwargs['network'] if 'network' in kwargs else None
+        self._use_private_ip = kwargs['use_private_ip'] if 'use_private_ip' in kwargs else None
```
Comment (mr-c) [inline src/toil/provisioners/gceProvisioner.py:141]:
```suggestion
        self._vpcSubnet = kwargs.get('vpcSubnet', None)
        self._network = kwargs.get('network', None)
        self._use_private_ip = kwargs.get('use_private_ip', None)
```

File: src/toil/utils/toilLaunchCluster.py [hunk 3]
```diff
@@ -63,9 +63,14 @@ def main() -> None:
                              "      \"Name\": clusterName,\n"
                              "      \"Owner\": IAM username\n"
                              " }. ")
+
+    parser.add_argument("--network",
+                        help="cloud network to use. default: 'default'")
```
Comment (mr-c) [inline src/toil/utils/toilLaunchCluster.py:68]:
```suggestion
                        help="GCE cloud network to use. default: 'default'")
```

**Pull request merged**